pygeos/constructive.py

```diff
@@ -703,3 +704,33 @@ def voronoi_polygons(
     <pygeos.Geometry GEOMETRYCOLLECTION EMPTY>
     """
     return lib.voronoi_polygons(geometry, tolerance, extend_to, only_edges, **kwargs)
+
+
+@requires_geos("3.6.0")
+@multithreading_enabled
+def minimum_rotated_rectangle(geometry, **kwargs):
```

Comment:
In PyGEOS we try to stick to the PostGIS namings of the functions. This one is named `ST_OrientedEnvelope` in postgis, so I'd like to propose `oriented_envelope` here.

Reply:
I've changed it to `oriented_envelope`. Just out of curiosity, how will this play together with shapely's (different) names once it is merged together?

Reply:
Yeah, I just wanted to comment here: Shapely already had this as `minimum_rotated_rectangle`, so we will need to keep that as an alias anyway, if we want to have the Postgis-compatible name as well

Reply:
Also, another argument for `minmum_rotated_rectangle` name is the consistentcy with `minimum_bounding_circle`

Reply:
I can switch back to `minimum_rotated_rectangle` as a default function, which is consistent with GEOS terminology (hence easier to follow through the code) and add `oriented_envelope` as an alias. Unless you want to do this once this will go to shapely.

Reply:
> Also, another argument for minmum_rotated_rectangle name is the consistentcy with minimum_bounding_circle

And an argument for `oriented_envelope` is consistency with `envelope`. Neither solution is objectively better than the other. 

Naming discussions always take a ridiculous amount of time away from the development process. I intended to prevent this from happening in pygeos by blindly sticking to an already present corpus, namely, postgis. I don't see a reason for deviating from this now.

Because of the future merge with shapely, I think an alias is a good idea. But then `minimum_rotated_rectangle` can be the alias.

Reply:
> Because of the future merge with shapely, I think an alias is a good idea. But then minimum_rotated_rectangle can be the alias.

Okay, I've done `minimum_rotated_rectangle` as an alias.